src/coreclr/jit/lower.h

```diff
@@ -465,6 +467,37 @@ class Lowering final : public Phase
     // Makes 'childNode' contained in the 'parentNode'
     void MakeSrcContained(GenTree* parentNode, GenTree* childNode) const;
 
+    // Makes 'childNode' regOptional in the 'parentNode'
+    void MakeSrcRegOptional(GenTree* parentNode, GenTree* childNode) const;
+
+    // Tries to make 'childNode' contained or regOptional in the 'parentNode'
+    inline void TryMakeSrcContainedOrRegOptional(GenTree* parentNode, GenTree* childNode) const
+    {
+#if defined(FEATURE_HW_INTRINSICS)
+        // HWIntrinsic nodes should use TryGetContainableHWIntrinsicOp and its relevant handling
+        assert(!parentNode->OperIsHWIntrinsic());
+#endif
```

Comment:
Does this need to be under an `#ifdef`? I think `OperIsHWIntrinsic` is available in non-HWI builds as well.

Reply:
Fixed.